app/commands/exercise/representation/create_or_update.rb

```diff
@@ -6,13 +6,26 @@ class CreateOrUpdate
       initialize_with :submission, :ast, :ast_digest, :mapping, :last_submitted_at
 
       def call
+        # First cache the old representation
+        # TODO: This needs tests adding
+        old_source_submission_representation = Exercise::Representation.find_by(source_submission: submission)
```

Comment:
Will there be only one?

Reply:
No. That's an excellent point. It needs to be the most recent. @ErikSchierboom I'll leave this one for you to update too.